src/mme-app/handlers/attach_csresp.c

```diff
@@ -134,6 +134,15 @@ stage6_processing()
 
 	memcpy(&(ue_entry->pdn_addr), &(csr_info->pdn_addr),
 		sizeof(struct PAA));
+        memcpy(&(ue_entry->pdn_addr), &(csr_info->pdn_addr),
+                sizeof(struct PAA));
+
+    char imsi[16] = {0};
+    struct in_addr paa=ue_entry->pdn_addr.ip_type.ipv4;
+    paa.s_addr = ntohl(paa.s_addr);
+    imsi_bin_to_str(ue_entry->IMSI, imsi);
+    log_msg(LOG_DEBUG, "IMSI %s - IP address assigned %s", 
```

Comment:
log_msg needs to be turned into macro. and imsi_bin_to_Str conversion should be done only if logging is enabled. Else its costly operation.

Reply:
TBD. Will open task for it.